sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreMap.scala

```diff
@@ -129,7 +129,11 @@ class PrefixScannableHDFSBackedStateStoreMap(
     other match {
       case o: PrefixScannableHDFSBackedStateStoreMap =>
         map.putAll(o.map)
-        prefixKeyToKeysMap.putAll(o.prefixKeyToKeysMap)
```

Comment:
This leads the prefix key maps across versions to "share" the values which it shouldn't. (especially the case of rolling back)

Reply:
Let's add one comment here.

Reply:
Thanks! Addressed via 7913e381e36751e61e405d22dbb1418b00a4135b